### PR TITLE
NF: MoveFileOrDirectoryTest consistent with other Move

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I know explicitly state that in MoveFile and MoveDirectory, the destination is
the actual new path of the file/directory. (And not the directory where the
copied element will be)

For the sake of consistency, I change MoveFileOrDirectory so that its
destination is also the new path (and not the path of the directory containing
the copied file).

I discovered it while trying to make more specific test, where the
source/destination of elements were asserted. Those test are now committed too.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveDirectory.kt [hunk 1]
```diff
@@ -18,9 +18,18 @@ package com.ichi2.anki.servicelayer.scopedstorage
 
 import androidx.annotation.VisibleForTesting
 import com.ichi2.anki.model.Directory
+import com.ichi2.anki.servicelayer.scopedstorage.MigrateUserData.Operation
 import timber.log.Timber
 import java.io.File
 
+/**
+ * [Operation] which safely moves a directory at path `source` to path `destination`.
+ * `destination is the new path of the directory.
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveDirectory.kt:27]:
```suggestion
 * `destination` is the new path of the directory.
```

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt [hunk 2]
```diff
@@ -30,6 +30,9 @@ import java.io.File
  *  Checking whether a `File` object is a file/directory requires an I/O operation, which means that
  *  it should not be done in a loop, as this would block preemption.
  *
+ *  @param sourceFile is the file or directory to move
+ *  @param destination the new path of the file or directory (not the directory containing it)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt:34]: I believe these should be moved to being above the parameters themselves (my Kotlin could be wrong here)

**Review by david-allison — COMMENTED** [review]
Two nitpicks. Conceptually looks fine. Treat this as a soft-approve

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt:34]: Not according to https://kotlinlang.org/docs/kotlin-doc.html#block-tags
This document a property of the constructor

**Review by Arthur-Milchior — COMMENTED** [review]
Please squash when merging

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFileOrDirectory.kt:34]: I stand corrected, thanks!

**Review by david-allison — APPROVED** [review]
LGTM

**Comment by mikehardy** [comment]
#5304

**Pull request merged**